src/smpi/instr_smpi.cpp

```diff
@@ -433,3 +450,113 @@ void TRACE_smpi_recv(int rank, int src, int dst, int tag)
   XBT_DEBUG("Recv tracing from %d to %d, tag %d, with key %s", src, dst, tag, key);
   new_pajeEndLink (SIMIX_get_clock(), PJ_container_get_root(), type, container, "PTP", key);
 }
+
+/**************** Functions to trace the migration of tasks. *****************/
+
+void TRACE_smpi_task_migrate(int rank, sg_host_t host)
+{
+  if (!TRACE_smpi_is_enabled()) return;
+  
+  char str[INSTR_DEFAULT_STR_SIZE];
+  snprintf(str, INSTR_DEFAULT_STR_SIZE, "%s-rank-%d", sg_host_get_name(host), rank);
+  
+  if(PJ_container_get_or_null(str)){
+    return;
+  }
+
+  container_t father;
+  if (TRACE_smpi_is_grouped()){
+    father = PJ_container_get(sg_host_get_name(host));
+  }else{
+    father = PJ_container_get_root();
+  }
+  xbt_assert(father!=NULL,
+	      "Could not find a parent for mpi rank %s at function %s", str,
+	      __FUNCTION__);
+  PJ_container_new(str, INSTR_SMPI, father);
+}
+
+long long int get_migration_counter()
+{
```

Comment:
I am surprised to see that such code actually works for you. This is grossly non reentrant, so you'll need to add a comment explaining which assumption you make on the caller's behavior. Who is the caller, and how you know that calls to this function do not intermix when they are several migrations at the same time.